src/main/java/core/basesyntax/service/FruitServiceImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.Fruit;
+import java.util.Map;
+
+public class FruitServiceImpl implements FruitService {
+
+    @Override
+    public String getReport() {
```

Comment:
Potentially, it's better not to use internal storage, so we're hardwired to the class and it can't be used for anything else. Processing in the same way, only take the map not from storage, but pass it as a parameter to the method, but when calling, pass it from storage.